File: Cpp/551_Student_Attendance_Record_I.cpp
class Solution {
public:
    bool checkRecord(string s) {
        int absent_count = 0;
        int late_count = 0;
        for(int i=0; i<s.size(); ++i){
            if(s[i] == 'A') {
                ++absent_count;
                if(absent_count > 1) return false;
            } else if(s[i] == 'L'){
                ++late_count;
                if(late_count > 2) return false;
                continue;
            }
            late_count = 0;
        }
        
        return true;
    }
};

File: Kotlin/551_Student_Attendance_Record_I.kt
class Solution {
    fun checkRecord(s: String): Boolean {
        var absentCount = 0
        var lateCount = 0
        
        for(ch in s){
            if(ch == 'A') {
                ++absentCount
                if(absentCount > 1) return false
            } else if(ch == 'L'){
                ++lateCount
                if(lateCount > 2) return false
                continue
            }
            
            lateCount = 0
        }
        
        return true
    }
}